### PR TITLE
New proof script language

### PR DESCRIPTION
**Pull request by mattulbrich**
The old script engine has lots of unneeded code from proof script debugger for KeY.

### CODE REVIEW

**Comment by JonasKlamroth** [comment]
i hope i fixed most of the NPEs. They came basically all from the same problem that node.getChildren may return null. I dont know if that is the intended behaviour or if it is more intuitive to just return an empty list. @mattulbrich what are your thoughts on that? the proofnodecheckpoints creation needs further adaptation but a rudimentary solution is working now. I could look into that if this PR is approved in general

File: modules/core/src/edu/kit/iti/algover/proof/ProofNode.java [hunk 1]
```diff
@@ -131,171 +105,69 @@ public SymbolTable getAllSymbols() {
...
-        return variableAssignments;
-    }
-
-    /**
-     * non-obligatory label for handling cases
-     */
-    public String getLabel() {
-        return label;
+    public List<ProofNode> getChildren() {
+        return children == null ? null : Collections.unmodifiableList(children);
```
Comment (JonasKlamroth) [inline modules/core/src/edu/kit/iti/algover/proof/ProofNode.java:167]: this is the line i mentioned in my comment. do we rly want to return null here?

Comment (mattulbrich) [reply modules/core/src/edu/kit/iti/algover/proof/ProofNode.java:176]: Yes

`getChildren() == null` means no rule applied or error during rule application

`getChildren().isEmpty()` means that a closing rule has been successfully applied.

Of course, we could use a different kind of mechanism ...

Comment (mattulbrich) [reply modules/core/src/edu/kit/iti/algover/proof/ProofNode.java:175]: Moreover, I discovered that I implemented an alternative method `getSuccessors()` never returning `null` but the empty list instead.

**Pull request merged**